flink-connector-pinot/src/test/java/org/apache/flink/streaming/connectors/pinot/PinotTestBase.java

```diff
@@ -0,0 +1,262 @@
+                                }
+                                return false;
+                            })
+                            // Allow Pinot to take up to 180s for starting up
+                            .withStartupTimeout(Duration.ofSeconds(180))
+            );
+
+    /**
+     * Creates a new instance of the {@link PinotTestHelper} using the testcontainer port mappings
+     * and creates the test table.
+     *
+     * @throws IOException
+     */
+    @BeforeEach
+    public void setUp() throws IOException {
+        pinotHelper = new PinotTestHelper(getPinotHost(), getPinotControllerPort(), getPinotBrokerPort());
+        pinotHelper.createTable(TABLE_CONFIG, TABLE_SCHEMA);
+    }
+
+    /**
+     * Delete the test table after each test.
+     *
+     * @throws Exception
+     */
+    @AfterEach
+    public void tearDown() throws Exception {
+        pinotHelper.deleteTable(TABLE_CONFIG, TABLE_SCHEMA);
+    }
+
+    /**
+     * Returns the host the Pinot container is available at
+     *
+     * @return Pinot container host
+     */
+    protected String getPinotHost() {
+        return this.pinot.getHost();
+    }
+
+
+    /**
+     * Returns the Pinot controller port from the container ports.
+     *
+     * @return Pinot controller port
+     */
+    protected String getPinotControllerPort() {
+        return this.pinot.getMappedPort(PINOT_INTERNAL_CONTROLLER_PORT).toString();
+    }
+
+    /**
+     * Returns the Pinot broker port from the container ports.
+     *
+     * @return Pinot broker port
+     */
+    protected String getPinotBrokerPort() {
+        return this.pinot.getMappedPort(PINOT_INTERNAL_BROKER_PORT).toString();
+    }
+
+    /**
+     * Class defining the elements passed to the {@link PinotSink} during the tests.
+     */
+    protected static class SingleColumnTableRow {
+
+        private String _col1;
+        private Long _timestamp;
+
+        SingleColumnTableRow(@JsonProperty(value = "col1", required = true) String col1,
+                             @JsonProperty(value = "timestamp", required = true) Long timestamp) {
+            this._col1 = col1;
+            this._timestamp = timestamp;
+        }
+
+        @JsonProperty("col1")
+        public String getCol1() {
+            return this._col1;
+        }
+
+        public void setCol1(String _col1) {
+            this._col1 = _col1;
+        }
+
+        @JsonProperty("timestamp")
+        public Long getTimestamp() {
+            return this._timestamp;
+        }
+
+        public void setTimestamp(Long timestamp) {
+            this._timestamp = timestamp;
+        }
+    }
+
+
+    /**
+     * EventTimeExtractor for {@link SingleColumnTableRow} used in e2e tests.
+     * Extracts the timestamp column from {@link SingleColumnTableRow}.
+     */
+    protected static class SingleColumnTableRowEventTimeExtractor extends EventTimeExtractor<SingleColumnTableRow> {
```

Comment:
Move to `PinotSinkTest` because it is only used there?

Reply:
As the `SingleColumnTableRowEventTimeExtractor` will only be used within sink tests it makes sense to move it there.